### ISSUE TITLE
many_buttons regression

### ISSUE THREAD

**Issue by DGriffin91** [bevyengine/bevy#16318]
## Bevy version

This regression was introduced in Remove custom rounding (#16097) at 05d90686fc355c6ae43b108e84b0a7a0a01d7bf9

The commit just before it: af279073aa89352ee7ded402a943f2168e37ef10 is faster.

`cargo run --release --example many_buttons`
Before the #16097 PR: `17.6ms`, after: `21.7ms`.

Windows 10, 7950x, RTX3060, Vulkan

After the PR (05d90686fc355c6ae43b108e84b0a7a0a01d7bf9)
![Image](https://github.com/user-attachments/assets/15edecd8-5b64-450c-a397-1cf404891cff)

Before the PR (af279073aa89352ee7ded402a943f2168e37ef10)
![Image](https://github.com/user-attachments/assets/0ea55d69-f553-4207-a4cb-46b4e5aabd03)

vtune with `cargo run --profile release-with-debug --example many_buttons`

```
[profile.release-with-debug]
inherits = "release"
debug = true
```

**Comment by ickshonpe** [comment]
Yep, Taffy's custom rounding is quite a bit slower as it walks the node tree another time after generating the UI layout, whereas the custom rounding code that was removed applied the rounding during the walk to update all the nodes. 

Maybe we could copy Taffy's rounding into `bevy_ui_layout` system, but it's a bit fragile and overcomplicated already which is why I didn't do that in #16097.
